services/core/Ambient/tests/test_ambient_agent.py

```diff
@@ -0,0 +1,344 @@
+                                                        "Ambient weather API "
+                                                        "key needs to be set in "
+                                                        "the test_ambient_data.json"
+                                                        " variable api_key")
+pytestmark_app = pytest.mark.skipif(not APP_KEY, reason="No APP key found. "
+                                                        "Ambient weather application"
+                                                        " key needs to be set in "
+                                                        "the test_ambient_data.json"
+                                                        " variable app_key")
+pytestmark_locs = pytest.mark.skipif(SKIP_LOCS, reason="Invalid locations list for"
+                                                       "Ambient agent. Ambient "
+                                                       "agent requires a list of "
+                                                       "location strings "
+                                                       "corresponding to devices "
+                                                       "owned/operated by the test"
+                                                       "runner to be set in  the "
+                                                       "test_ambient_data.json"
+                                                       " variable app_key")
+
+
+@pytest.fixture(scope="module")
+def query_agent(request, volttron_instance):
+    # 1: Start a fake agent to query the historian agent in volttron_instance
+    agent = volttron_instance.build_agent()
+    agent.poll_callback = MagicMock(name="poll_callback")
+    # subscribe to weather poll results
+    agent.vip.pubsub.subscribe(
+        peer='pubsub',
+        prefix="weather/poll/current",
+        callback=agent.poll_callback).get()
+
+    # 2: add a tear down method to stop the fake
+    # agent that published to message bus
+    def stop_agent():
+        print("In teardown method of query_agent")
+        agent.core.stop()
+
+    request.addfinalizer(stop_agent)
+    return agent
+
+
+@pytest.fixture(scope="module", params=[ambient_service])
+def weather(request, volttron_instance):
+    print("** Setting up weather agent module **")
+    print("request param", request.param)
+    config = copy.copy(request.param)
+    source = config.pop('weather_service')
+    identity = config.pop('identity')
+
+    agent = volttron_instance.install_agent(
+        vip_identity=identity,
+        agent_dir=source,
+        start=False,
+        config_file=config)
+
+    volttron_instance.start_agent(agent)
+    gevent.sleep(3)
+
+    def stop_agent():
+        print("stopping weather service")
+        if volttron_instance.is_running():
+            volttron_instance.stop_agent(agent)
+
+    request.addfinalizer(stop_agent)
+    return agent, identity
+
+
+@pytest.fixture(scope="function")
```

Comment:
FYI, scope [defaults to `function`](https://docs.pytest.org/en/latest/reference.html#pytest-fixture); setting scope to `function` might be unnecessary.